R/combine_distance_matrices.R

```diff
@@ -0,0 +1,55 @@
+# combine_distance_matrices.R
+
+# This is a helper function for clusterSystems and is unlikely to be useful elsewhere.
+
+# Parameters :
+# mode                Method by which to combine distance matrices
+# distanceMatrices    List of objects of class distance.
+#
+# Value:
+# single object of class distance.
+#
+# Author: Rachel Silverstein
+
+combine_distance_matrices <- function(mode, distanceMatrices) {
```

Comment:
Can you refactor this to use `pmax()` and `pmin()` etc?
```
> x <- matrix(1:4, ncol=2)
> y <- matrix(4:1, ncol=2)
> x
     [,1] [,2]
[1,]    1    3
[2,]    2    4
> y
     [,1] [,2]
[1,]    4    2
[2,]    3    1
> pmax(x, y)
     [,1] [,2]
[1,]    4    3
[2,]    3    4
```

Reply:
I do use pmax and pmin, however I had to compare the matrices pairwise as I iterated through the list since pmax and pmin take vectors not lists. 

I could not figure out how to unlist the matrices without also flattening them, so I just opted for a loop.